packages/genesys-spark-components/src/components/stable/gux-tooltip-title/gux-tooltip-title.tsx

```diff
@@ -132,6 +136,7 @@ export class GuxTooltipTitle {
     this.getTitleElements().forEach(element => {
       if (element.tagName === 'GUX-ICON' && !this.getTitleTextContent()) {
         this.iconOnly = true;
+        this.addIconOnly();
```

Comment:
You don't need to introduce more state in the component here you can use the `iconOnly` property in the render function to toggle a class to get the same result.
```
  render(): JSX.Element {
    return [
      <span
        class={{
          'gux-title-container': true,
          'gux-icon-only': this.iconOnly,
        }}
      >
        <slot />
      </span>,
      this.renderTooltip()
    ] as JSX.Element;
  }
```
Obviously the scss will change a bit too.

Reply:
Much neater, thanks